xcp/compat.py

```diff
@@ -1,26 +1,43 @@
 """Helper module for setting up binary or UTF-8 I/O for Popen and open in Python 3.6 and newer"""
+# pyright: strict
+# pyright: reportTypeCommentUsage=false
+# pyright: reportUnknownParameterType=false
 # See README-Unicode.md for Details
 import sys
+from typing import TYPE_CHECKING
 
-if sys.version_info >= (3, 0):  # pragma: no cover
+if TYPE_CHECKING:
+    from typing import IO, Any  # pylint: disable=unused-import  # pragma: no cover
+
+# pylint: disable=unspecified-encoding
+if sys.version_info >= (3, 0):
     open_utf8 = {"encoding": "utf-8", "errors": "replace"}
 
-    def utf8open(filename, *args, **kwargs):
+    def open_with_codec_handling(filename, mode="r", encoding="utf-8", **kwargs):
+        # type:(str, str, str, Any) -> IO[Any]
         """Helper for open(): Handle UTF-8: Default to encoding="utf-8", errors="replace" for Py3"""
-        if "b" in (args[0] if args else kwargs.get("mode", "")):
+        if "b" in mode or "b" in kwargs.get("mode", ""):
```

Comment:
It could be omitted, but `mode` could either be passed (for example) as just `"rb"` or as `mode="rb"`.

To not ignore the latter, it would be better to keep checking kwargs for "b" as well.